_posts/2017-02-02-overlay-network-performance-testing.md

```diff
@@ -0,0 +1,75 @@
+---
+layout:     post
+title:      Overlay network performance testing
+date:       2017-02-02 10:00
+summary:    How do you decide on an overlay network to use for the next generation platform at SkyBet?
+category:   Infrastructure
+tags:       Networking, Performance, Containers, Docker, Kubernetes
+author:     martyn_ranyard
+---
+
+As one of the engineers working on our next-generation platform in the Bet tribe of **Sky Betting and Gaming**, I was tasked with choosing an overlay network for our container scheduler (we're going with Kubernetes but that's a story for a different post).  But just how do you choose an overlay network for a platform when you don't know exactly what will be running on it yet?  Here's how I went about it, some stats and some conclusions I came to.
+
+# What are the options?
+It turns out there are a wide variety of overlay networks to choose from and they all have different characteristics.  The current networking model for containers (e.g. Docker and Rkt - CoreOS' similar product) is called CNI and theoretically any overlay network that is CNI compliant can be used in Kubernetes.  The first wave of cutting down these networks is by ease of install - If I'm going to be testing networks en masse, I don't want to be spending days installing to find they're not suitable.
```

Comment:
CNI-compliant, not "CLI compliant".

`&mdash;`, not a hyphen, separating the phrases.

Needs another word in between "installing" and "to", probably something like "installing, only to find..."